src/adaptix/_internal/morphing/concrete_provider.py

```diff
@@ -75,6 +75,57 @@ def datetime_format_dumper(data: datetime):
         return datetime_format_dumper
 
 
+@dataclass
+@for_predicate(datetime)
+class DatetimeTimestampProvider(LoaderProvider, DumperProvider):
+    tz: typing.Optional[timezone] = timezone.utc
+
+    def _provide_loader(self, mediator: Mediator, request: LoaderRequest) -> Loader:
+        tz = self.tz
+
+        def datetime_timestamp_loader(data):
+            try:
+                if data is None:
+                    raise TypeLoadError(float, data)
+
+                return datetime.fromtimestamp(data, tz=tz)
+            except TypeError:
+                raise TypeLoadError(float, data)
+            except OverflowError:
+                raise ValueLoadError(
+                    "Timestamp is out of the range of values supported by the platform",
```

Comment:
This error message is for API clients, so "by platform" is reduntant here. It should be just "... of supported values"